assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.jsx

```diff
@@ -0,0 +1,176 @@
+import {
+  addDays,
+  addMonths,
+  addYears,
+  format,
+  parseISO,
+  setHours,
+  setMinutes,
+} from 'date-fns';
+import React, { useState } from 'react';
+import { noop, truncate } from 'lodash';
+import { EOS_CONTENT_COPY, EOS_INFO_OUTLINED } from 'eos-icons-react';
+import Button from '@common/Button';
+import Modal from '@common/Modal';
+import Input from '@common/Input';
+import Label from '@common/Label';
+import Select from '@common/Select';
+import Switch from '@common/Switch';
+
+const normalizeExpiration = (expiration) =>
+  setMinutes(setHours(expiration, 0), 0);
+
+const availableTimeOptions = [
+  {
+    type: 'months',
+    timeGenerator: (quantity) => addMonths(new Date(), quantity),
+  },
+  {
+    type: 'days',
+    timeGenerator: (quantity) => addDays(new Date(), quantity),
+  },
+  {
+    type: 'years',
+    timeGenerator: (quantity) => addYears(new Date(), quantity),
+  },
+];
+
+function ApiKeySettingsModal({
+  open = false,
+  loading = false,
+  onSave = noop,
+  onClose = noop,
+  generatedApiKey,
+  generatedApiKeyExpiration,
+}) {
+  const timeOptions = availableTimeOptions.map((o) => o.type);
+  const [timeQuantity, setTimeQuantity] = useState(0);
+  const [timeQuantityType, setTimeQuantityType] = useState(timeOptions[0]);
+
+  const [quantityError, setQuantityError] = useState(false);
+  const [timeFormEnabled, setTimeFormEnabled] = useState(true);
+
+  const saveFormState = () => {
+    if (!timeFormEnabled) {
+      onSave({ apiKeyExpiration: null });
+      return;
+    }
+    if (timeQuantity === 0 || !timeQuantity) {
+      setQuantityError(true);
+      return;
+    }
+    const timeQuantitySettings = availableTimeOptions.find(
+      (q) => q.type === timeQuantityType
+    );
+    const apiKeyExpiration = timeQuantitySettings.timeGenerator(timeQuantity);
+
+    onSave({ apiKeyExpiration: normalizeExpiration(apiKeyExpiration) });
+  };
+
+  return (
+    <Modal
+      title="API Key Settings"
+      className="!w-3/4 !max-w-3xl"
+      open={open}
+      onClose={onClose}
+    >
+      <div className="flex flex-col my-5">
+        <span className="my-2 mb-4 text-gray-500">
+          {' '}
+          By generating a new key, you will need to replace the API key on all
+          hosts.{' '}
+        </span>
+
+        <div className="flex space-x-1">
+          <div className="w-1/5">
+            <Label>Never Expires</Label>
+          </div>
+
+          <Switch
+            selected={!timeFormEnabled}
```

Comment:
Why this negation?
I would expect that if the `timeFormEnabled` is true the switch being selected

Reply:
Okey, forget about this. The logic is a negation in fact